test/repr/test_size_of.ml

```diff
@@ -0,0 +1,186 @@
+module T = Repr
+
+let encode_bin t = T.(unstage (encode_bin t))
+
+let encode_bin typ v =
+  let buffer = Buffer.create 0 in
+  encode_bin typ v (Buffer.add_string buffer);
+  Buffer.contents buffer
+
+let random_string len = String.init len (fun _ -> char_of_int (Random.int 256))
+
+let check_unknown ~__POS__:pos typ =
+  match T.Size.of_value typ with
+  | Unknown -> ()
+  | Dynamic _ | Static _ ->
+      Alcotest.failf ~pos
+        "Expected type to have unknown size, but (Dynamic _ | Static _) was \
+         received."
+
+let check_static ~__POS__:pos typ expected v =
+  match T.Size.of_value typ with
+  | Unknown | Dynamic _ ->
+      Alcotest.failf ~pos
+        "Expected type to have static size %d, but (Unknown | Dynamic _) was \
+         received."
+        expected
+  | Static n ->
+      Alcotest.(check ~pos int) "Expected static size" expected n;
+
+      (* Check that the encoding actually occupies [n] bytes *)
+      let actual_size = String.length (encode_bin typ v) in
+      Alcotest.(check ~pos int)
+        "Actual size must match static spec" expected actual_size
```

Comment:
You could add more checks here, similar to the ones in `check_dynamic`

Reply:
Good point; I've added a test that checks that the `of_encoding` sizer is static and matches the `of_value` one.